coalib/results/Diff.py

```diff
@@ -278,18 +276,37 @@ def modified(self):
         return result
 
     @property
+    def modified(self):
+        """
+        Calculates the modified file, after applying the Diff to the original.
+        Also adds linebreaks at the end of each line (except eof)
+        """
+        eof_linebreak = False
+
+        if self._file and self._file[-1].endswith('\n'):
+            eof_linebreak = True
+
+        file = [line.rstrip('\n') for line in self._modified()]
+
+        if eof_linebreak:
+            return self._add_linebreaks(file)
```

Comment:
> Assume a file like this: file = ['a\n', 'b']
> We delete the last line, resulting in file = ['a\n'].

I just meant this case. When a user supplies a text initially (i.e. via `__init__`) we shall preserve the the EOF-state. Only when we remove that line, we don't preserve state for "new last lines".